### ISSUE TITLE
SIGSEV just before "npar(antenna) =  -3.1774E+00 "

### ISSUE THREAD

**Issue by AreWeDreaming** [ORNL-Fusion/aorsa#34]
Hello,
I am trying to make use of the free NERSC time up for grabs at the moment. Last night I ran my first case with full ion absorption, but then noticed that my resolution was insufficient, afterward. 
So my case is as follow
448 x 384 grid size
96x64 nprow and npcol
256 nodes
The code crashes right after printing the "n   capr_bpol_mid" table.
The error message  is:
forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source             
xaorsa2d           000000002033FE13  for__signal_handl     Unknown  Unknown
libpthread-2.26.s  00002AAAB19F92D0  Unknown               Unknown  Unknown
libsci_intel_mpi.  00002AAAABE53EE6  blacs_barrier_        Unknown  Unknown
xaorsa2d           000000002020F486  Unknown               Unknown  Unknown
xaorsa2d           000000002000CC12  Unknown               Unknown  Unknown
libc-2.26.so       00002AAAB1C2934A  __libc_start_main     Unknown  Unknown
xaorsa2d           000000002000CB2A  Unknown

**Comment by AreWeDreaming** [comment]
I just checked the source code, the last line that we know that is executed is 4375 in eqdsk_setup.f.  I checked the MAXVMSIZE and it is small. Also, this run is still smaller than my first successful run.
AORSA worked when I was running it with a 256 by 256 grid with 32 by 32  columns on 128 Nodes. So there must be something wrong with my grid size or how it spread across the nodes. But I don't see anything wrong with what I am currently using.
I submitted a 384x384 grid on 64x64 cores on 128 Nodes. I will update you once I see how that goes.

**Comment by dlg0** [comment]
Hi @AreWeDreaming , could you point me to your run directory at NERSC? (and open its permissions so I can copy it and try this out myself?)

David.

**Comment by AreWeDreaming** [comment]
I am afraid I forgot to save the data of the crashed run before I started the new one, which is still computing. It might have been a memory issue this time. How do you diagnose these? I looked at the MaxVMEM but that was about one gigabyte. That seems very low. The job id of the crashed run is 34783109. I restored the submission files to the settings I used to produce the crash., and stored them at `/global/cfs/cdirs/m77/denks/170325_3850`. You should have read access there, right?
Since computation is free at the moment, and it only runs for 2 minutes, it might be worth reproducing. I am slightly hesitant to do so myself since I don't want to reduce my priority unnecessarily. I am still hopeful to use the fast turnover at available now at cori to improve my APS poster with some AORSA results.

**Comment by AreWeDreaming** [comment]
My run on 128 nodes finished but took a whopping 9 hours to run. Most of the time was spent computing the QL operators for the 4 ions species.
Since this run looked converged, I proceeded to submit the 10 more mode numbers. I wanted to speed up the process so I switched to npcol = 64. nprow = 60 and 480 nodes with 32 cores each. You can find an example output at: `/global/cfs/cdirs/m77/denks`. If this configuration is obviously wrong, please let me know quickly. There might be a window of opportunity here with CSCRSATCH being down because AORSA requires very little hard disk space for the amount of CPU time it uses.

**Comment by cornwalllau** [comment]
If this is the folder with 170325_3850_68, I don't see a completed run at all. The out15 and log_aorsa2d is mostly blank...

**Comment by AreWeDreaming** [comment]
Yes, `170325_3850_68` crashed just a couple of seconds in. I put my successful run `170325_3850_69` also into that folder for comparison. Thank you for the fast reply!

**Comment by cornwalllau** [comment]
Ah, I see it now. Taking a quick look, this run did finish and  is what I expected. Electron absorption is 99.84% and the ion absorption from the quasilinear wdot is really low and pretty much negligible. That's what I saw previously.

The run takes a long time because the Wdot calculation is quite memory intensive. Fred's solution at the time is to rerun wdot for each species, which of course increases the time. Wdot is therefore recalculated for each ion species. As you probably already noticed, each of the ion quasilinear Wdot takes up 2 hours each. 

Give that 99.84% of the power goes into electrons, Fred and I decided it did not make sense to run Wdot for the ion contribution part. There is therefore a tag in the input file file to separately calculate or not calculate the ion contribution. If you go to aorsa2din_mod.f90, you will notice the tags nzeta_wdote and nzeta_wdoti. I will suggest using nzeta_wdote  51 and nzeta_wdoti = 0.

**Comment by AreWeDreaming** [comment]
Ok, that might be a workaround, to cut the time down by 6 hours. I'll change the inputs and go back to 128 nodes. Thanks, Cornwall.
